### PR TITLE
remove the ability for pilot to register CRDs itself

### PR DESCRIPTION
**Pull request by ayj**
Installing and modifying CRDs requires elevated cluster-wide
permissions. This should be done by a separate install step
(istio-init) or by an operator.

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/pilot/templates/clusterrole.yaml [hunk 1]
```diff
@@ -8,24 +8,14 @@ metadata:
...
-  resources: ["*"]
-  verbs: ["get", "watch", "list"]
-- apiGroups: ["networking.istio.io"]
-  resources: ["*"]
-  verbs: ["*"]
-- apiGroups: ["authentication.istio.io"]
-  resources: ["*"]
-  verbs: ["*"]
-- apiGroups: ["apiextensions.k8s.io"]
-  resources: ["customresourcedefinitions"]
```
Comment (howardjohn) [inline install/kubernetes/helm/istio/charts/pilot/templates/clusterrole.yaml]: do we not need `get` for anything? I don't think we do but just checking

**Review by howardjohn — APPROVED** [review]
lgtm

File: install/kubernetes/helm/istio/charts/pilot/templates/clusterrole.yaml (hunk 1 above)
Comment (ayj) [reply install/kubernetes/helm/istio/charts/pilot/templates/clusterrole.yaml]: Pilot doesn't. Istiod will eventually need this when Galley's processing pipeline is enabled.

**Comment by ayj** [comment]
The controller_test.go is depending on the mock type being created. Fix pending.

**Pull request merged**